app/models/report.rb

```diff
@@ -51,6 +51,14 @@ def destroy_reported_item
     mark_as_reviewed
   end
 
+  def mark_as_nsfw
```

Comment:
>setting the profile NSFW

This should only be done if the majority of the user's content is NSFW. I know people complaining about that for only one post of a user who definitely not deserves to be checked as NSFW.

IMO the easiest solution is to check manually: if the post really is porn related or whatever, then delete the post (the author is not compliant with the ToS), if it's not (just erotic or something like that), then ignore the request. It's not longer than to turn the post to nsfw anyway and I agree that inconsistency in the network is bad.